### PR TITLE
Events/docs

### PR DESCRIPTION
**Pull request by drewbailey**
api docs for `/v1/event/stream`

### CODE REVIEW

File: website/pages/api-docs/events.mdx [hunk 1]
```diff
@@ -0,0 +1,626 @@
+---
+layout: api
+page_title: Events - HTTP API
+sidebar_title: Events
+description: |-
+  The /event endpoints interact with the local Nomad agent to interact with
+  members and servers.
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:7]: How about something like, "The /event endpoints are used to query for and interact with Nomad events"

File: website/pages/api-docs/events.mdx [hunk 2]
```diff
@@ -0,0 +1,626 @@
...
+---
+
+# Events HTTP API
+
+The `/event` endpoints are used to query for recent events.
+
+## Event Stream
+
+This endpoint streams a servers backlog off events as we as new events as they occur.
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:16]:
```suggestion
This endpoint streams a servers backlog of events as well as new events as they occur.
```

File: website/pages/api-docs/events.mdx [hunk 3]
```diff
@@ -0,0 +1,626 @@
...
+The stream will be kept alive until the connection is closed.
+
+| Method | Path               | Produces           |
+| ------ | ------------------ | ------------------ |
+| `GET`  | `/v1/event/stream` | `application/json` |
+
+
+The Table below shows this endpoint's [required ACLs](/api-docs#acls). Due to
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:24]:
```suggestion
The table below shows this endpoint's [required ACLs](/api-docs#acls). Due to
```

File: website/pages/api-docs/events.mdx [hunk 4]
```diff
@@ -0,0 +1,626 @@
...
+the nature of this endpoint different topics require different policies.
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:25]:
```suggestion
the nature of this endpoint individual topics require specific policies.
```

File: website/pages/api-docs/events.mdx [hunk 5]
```diff
@@ -0,0 +1,626 @@
...
+| `Job`        | `namespace:read-job` |
+| `Alloc`      | `namespace:read-job` |
+| `Deployment` | `namespace:read-job` |
+| `Eval`       | `namespace:read-job` |
+| `Node`       | `node:read`          |
+
+### Parameters
+
+- `index` `(int: 0)` - Specifies the index to start streaming events from. If
+  the requeste index is no longer in the buffer the stream will start at the
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:39]:
```suggestion
  the requested index is no longer in the buffer the stream will start at the
```

File: website/pages/api-docs/events.mdx [hunk 6]
```diff
@@ -0,0 +1,626 @@
...
+  nearest index.
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:40]:
```suggestion
  next available index.
```

File: website/pages/api-docs/events.mdx [hunk 7]
```diff
@@ -0,0 +1,626 @@
...
+
+- `topic` `(topic:filter_key: "*:*")` - Specifies a topic to subscribe to and
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:42]:
```suggestion
- `topic` `(topic:filter_key: "*:*")` - Specifies the set of topics to subscribe to and
```

File: website/pages/api-docs/events.mdx [hunk 8]
```diff
@@ -0,0 +1,626 @@
...
+  filter on. By default all topics are subscribed to. Multiple topics should be
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:43]:
```suggestion
  filter on. The default is to subscribe to all topics. Multiple topics may be
```

File: website/pages/api-docs/events.mdx [hunk 9]
```diff
@@ -0,0 +1,626 @@
...
+  specified by passing multiple `topic` parameters. A valid topic parameter
+  includes a `topic` type and an optional `filter_key` separated by a colon
+  `:`. As an example `?topic=Deployment:redis` would subscribe to all
+  `Deployment` events for a job redis. To only subscribe to `Node` events a
```
Comment (shoenig) [inline website/pages/api-docs/events.mdx:47]: Can there be multiple filters per topic?

File: website/pages/docs/configuration/server.mdx [hunk 10]
```diff
@@ -65,6 +68,9 @@ server {
   [encryption documentation][encryption] for more details on this option
   and its impact on the cluster.
 
+- `event_buffer_size` `(int: 100)` - Specifies the amount of events generated
```
Comment (shoenig) [inline website/pages/docs/configuration/server.mdx:71]:
```suggestion
- `event_buffer_size` `(int: 100)` - Specifies the number of events generated
```

File: website/pages/docs/configuration/server.mdx [hunk 11]
```diff
@@ -54,6 +54,9 @@ server {
   this server will handle. This can be used to restrict the evaluations that
   worker threads will dequeue for processing.
 
+- `enable_event_broker` `(bool: true)` - Specifies if this server's state store
```
Comment (shoenig) [inline website/pages/docs/configuration/server.mdx:57]:
```suggestion
- `enable_event_broker` `(bool: true)` - Specifies if this server
```

Mentioning the state store seems like an implementation detail - could we just say the server is doing this?

File: website/pages/docs/configuration/server.mdx [hunk 12]
```diff
@@ -65,6 +68,9 @@ server {
   [encryption documentation][encryption] for more details on this option
   and its impact on the cluster.
 
+- `event_buffer_size` `(int: 100)` - Specifies the amount of events generated
+  by the server's state store to be held in memory.
```
Comment (shoenig) [inline website/pages/docs/configuration/server.mdx:72]:
```suggestion
  by the server to be held in memory. 
```

likewise

File: website/pages/docs/configuration/server.mdx [hunk 13]
```diff
@@ -90,6 +96,11 @@ server {
   deployment must be in the terminal state before it is eligible for garbage
   collection. This is specified using a label suffix like "30s" or "1h".
 
+- `durable_event_count` `(int: 100)` - Specifies the amount of events to
```
Comment (shoenig) [inline website/pages/docs/configuration/server.mdx:99]:
```suggestion
- `durable_event_count` `(int: 100)` - Specifies the number of events to
```

File: website/pages/docs/configuration/server.mdx [hunk 14]
```diff
@@ -90,6 +96,11 @@ server {
...
+  persist during snapshot generation. This provides a given servers event
```
Comment (shoenig) [inline website/pages/docs/configuration/server.mdx:100]:
```suggestion
  persist during snapshot generation. This provides a given server's event
```

File: website/pages/docs/configuration/server.mdx [hunk 15]
```diff
@@ -90,6 +96,11 @@ server {
...
+  stream durability during restarts by persiting a backlog of events to be
+  repopulated when the server comes back online.
```
Comment (shoenig) [inline website/pages/docs/configuration/server.mdx:102]: Should we comment on why one might increase or decrease this value?

File: website/pages/docs/configuration/server.mdx (hunk 10 above)
Comment (shoenig) [reply website/pages/docs/configuration/server.mdx:71]: Should we comment on why one might increase or decrease this value?

**Review by shoenig — APPROVED** [review]
Some comments / suggestions, overall looks good!

File: website/pages/api-docs/events.mdx (hunk 1 above)
Comment (drewbailey) [reply website/pages/api-docs/events.mdx:7]: :grimacing: woops, that's a bad copy paste :sweat_smile:

File: website/pages/api-docs/events.mdx (hunk 9 above)
Comment (drewbailey) [reply website/pages/api-docs/events.mdx:47]: yes will add to this to reflect

File: website/pages/docs/configuration/server.mdx (hunk 10 above)
Comment (drewbailey) [reply website/pages/docs/configuration/server.mdx:71]: yeah, how about 
```
- `event_buffer_size` `(int: 100)` - Specifies the number of events generated
  by the server's state store to be held in memory. Increasing this value
  enables new subscribers to have a larger lookback window when initially
  subscribing. Decreasing will lower the amount of memory used for the event
  buffer.
```

**Pull request merged**